### DISCUSSION TITLE
Limit switch causing lockup or reset

### DISCUSSION THREAD

**Discussion by Squeack** [terjeio/grblHAL#355] [Q&A]
I had a working 3-axis CNC mill driven by GrblHAL, except the X axis would occasionally completely stop reponding. I suspect it was a hardware error within cheap Chinese S109AFTG (aka 'enhanced' TB6600) drivers. I replaced those with DM556T drivers - still Chinese but from the reputable StepperOnline. This required redoing the wiring between the Teensy Breakout board (one of Phil's) and the drivers as the connections are in a different order, but the limit switch connections were not touched. The steppers all turn when instructed, and are now much quieter, but there seems to be a slight start-short pause-continue smoothly to the motion, which may or may not be related to the main problem.
The main problem comes from the limit switches. When these are triggered, either the Teensy goes unresponsive, or resets. I would guess this to be a power issue, perhaps supply somehow being briefly shorted to ground, but this is just a guess.
Does anybody have any suggestions on where to look? The wiring looks correct, having checked multiple times, I have attached the console output during two occurancies - one lockup and one reset.

[GRBLoutput_during_lockup.txt](https://github.com/terjeio/grblHAL/files/7890129/GRBLoutput_during_lockup.txt)

**Comment by terjeio** [comment]
Output looks ok - but the feed hold input somehow gets triggered.

`[DRIVER VERSION:210617]`

is an old version - [please upgrade](https://github.com/grblHAL/iMXRT1062). There is a bug in the framework code that can cause strange behaviour that has been fixed in newer versions.
Note that this repository is no longer maintained and I will archive it soon, if you have issues after an upgrade please reopen at the new one.

**Reply by Squeack** [reply]
Thank you. Will try the new version in the next couple of days and reraise if necessary.

**Reply by phil-barrett** [reply]
Newer prebuilt binaries are in grblHAL-teensy-4.x/prebuilt.md in my GitHub repo.  I would link directly but I'm away from my PC and the phone isn't cooperating. Will be back late today, let me know if you are still having trouble.
